src/meta/src/hummock/vacuum.rs

```diff
@@ -112,6 +113,15 @@ impl VacuumManager {
         let mut batch_idx = 0;
         let batch_size = 500usize;
         let mut sent_batch = Vec::with_capacity(objects_to_delete.len());
+        if self.env.opts.enable_hummock_data_archive {
```

Comment:
Prior to this PR, when `enable_hummock_data_archive = true`, stale objects cleaned by compaction will be deleted on full gc. IIUC in this PR, these objects will never be deleted. To maintain the previous behavior, should we skip [`mark_objects_for_deletion`](https://github.com/risingwavelabs/risingwave/blob/7bc3c9268e89af4d0a3c0768dedcadef8e37246b/src/meta/src/hummock/manager/versioning.rs#L117) in version checkpoint instead?